contrib/mesos/pkg/executor/service/service.go

```
@@ -86,188 +76,164 @@ func (s *KubeletExecutorServer) AddFlags(fs *pflag.FlagSet) {
-		EventRecordQPS:          s.EventRecordQPS,
-		FileCheckFrequency:      s.FileCheckFrequency,
-		HostnameOverride:        s.HostnameOverride,
-		HostNetworkSources:      hostNetworkSources,
-		HostPIDSources:          hostPIDSources,
-		HostIPCSources:          hostIPCSources,
-		// HTTPCheckFrequency
-		ImageGCPolicy: imageGCPolicy,
-		KubeClient:    apiclient,
-		// ManifestURL: ""
-		ManifestURLHeader:         manifestURLHeader,
-		MasterServiceNamespace:    s.MasterServiceNamespace,
-		MaxContainerCount:         s.MaxContainerCount,
-		MaxOpenFiles:              s.MaxOpenFiles,
-		MaxPerPodContainerCount:   s.MaxPerPodContainerCount,
-		MaxPods:                   s.MaxPods,
-		MinimumGCAge:              s.MinimumGCAge,
-		Mounter:                   mounter,
-		NetworkPluginName:         s.NetworkPluginName,
-		NetworkPlugins:            app.ProbeNetworkPlugins(s.NetworkPluginDir),
-		NodeStatusUpdateFrequency: s.NodeStatusUpdateFrequency,
-		OOMAdjuster:               oomAdjuster,
-		OSInterface:               kubecontainer.RealOS{},
-		PodCIDR:                   s.PodCIDR,
-		PodInfraContainerImage:    s.PodInfraContainerImage,
-		Port:              s.Port,
-		ReadOnlyPort:      s.ReadOnlyPort,
-		RegisterNode:      s.RegisterNode,
-		RegistryBurst:     s.RegistryBurst,
-		RegistryPullQPS:   s.RegistryPullQPS,
-		ResolverConfig:    s.ResolverConfig,
-		ResourceContainer: s.ResourceContainer,
-		RootDirectory:     s.RootDirectory,
-		Runonce:           s.RunOnce,
-		// StandaloneMode: false
-		StreamingConnectionIdleTimeout: s.StreamingConnectionIdleTimeout,
-		SyncFrequency:                  s.SyncFrequency,
-		SystemContainer:                s.SystemContainer,
-		TLSOptions:                     tlsOptions,
-		VolumePlugins:                  app.ProbeVolumePlugins(),
-		Writer:                         writer,
+	if err != nil {
+		// required for k8sm since we need to send api.Binding information back to the apiserver
+		log.Fatalf("No API client: %v", err)
```

let's return an error here instead